### PR TITLE
Updated internal variable names to match that of datapackage #860

### PR DESCRIPTION
**Pull request by jainamritanshu**
Updated internal variable names to match that of datapackage spec #765
The following changes were done for the variable names

```
tags -> keywords
nulls -> missingValues
name -> title
shortname -> name

The changes were done in the following files -

retriever/lib/compile.py
retriever/lib/datapackage.py
retriever/lib/engine.py
retriever/lib/parse_script_to_json.py
retriever/lib/templates.py
retriever/lib/tools.py
scripts/bioclim.py
scripts/biomass_allometry_db.py
scripts/breed_bird_survey.py
scripts/breed_bird_survey_50stop.py
scripts/forest_inventory_analysis.py
scripts/gentry_forest_transects.py
scripts/npn.py
scripts/plant_life_hist_eu.py
scripts/prism_climate.py
scripts/vertnet.py
scripts/wood_density.py
scripts/*.json(almost all datapackages) transition missingValues -> missing_values
test/test_retriever.py
retriever/__main__.py
```

@henrykironde I have made the changes after updating it with the master branch

### CODE REVIEW

File: scripts/intertidal_abund_me.py [hunk 1]
```diff
@@ -14,8 +15,12 @@ def __init__(self):
                         retriever_minimum_version='2.0.dev',
                         version='1.4.0',
                         urls={"main": "https://ndownloader.figshare.com/files/5600831"},
-                        tables={"main": Table("main", cleanup=Cleanup(correct_invalid_value, nulls=[-999.9]))}
+                        tables={"main": Table("main", cleanup=Cleanup(correct_invalid_value, missingValues=[-999.9]))}
```
Comment (henrykironde) [inline scripts/intertidal_abund_me.py]: we can clean up this file to a structure like 
```
class main(Script):
    def __init__(self, **kwargs):
        Script.__init__(self, **kwargs)
        self.name =
```
so that we take care of this clean up function like we have done in other files
`cleanup=Cleanup(correct_invalid_value, missingValues=[-999.9]))}`

Comment (jainamritanshu) [inline scripts/intertidal_abund_me.py]: @henrykironde Sure, I somehow missed this one. Will make the changes right now

File: scripts/plant_life_hist_eu.py [hunk 2]
```diff
@@ -68,7 +68,7 @@ def download(self, engine=None, debug=False):
             new_data.close()
             self.engine.auto_create_table(Table(key,
                                                 cleanup=Cleanup(correct_invalid_value,
-                                                                nulls=[-999.9])),filename=str("new" + key))
+                                                                missingValues=[-999.9])),filename=str("new" + key))
```
Comment (henrykironde) [inline scripts/plant_life_hist_eu.py]: what about the clean up for this?

Comment (jainamritanshu) [inline scripts/plant_life_hist_eu.py]: @henrykironde Sorry I seem to have missed a couple of scripts. I will fix this asap

**Comment by henrykironde** [comment]
looks really clean, and remember to commit --amend so that we keep only two commits.

**Comment by ethanwhite** [comment]
This isn't a full review, but I noticed that the current version test is:

```
if parse_version(VERSION) < parse_version("2.0.0"):
```

This should be `<=` since it is providing backward compatibility for `2.0.0`.

Along with this change we should go ahead and bump the version number (we should have done this right after the release anyway) to `2.1.dev`

**Comment by jainamritanshu** [comment]
@ethanwhite I have made the required changes

File: scripts/intertidal_abund_me.py (hunk 1 above)
Comment (henrykironde) [inline scripts/intertidal_abund_me.py]: in this file you did not updated internal variable names

**Comment by henrykironde** [comment]
In some files we dd not `import parse_version`

File: scripts/breed_bird_survey_50stop.py [hunk 3]
```diff
@@ -89,10 +100,9 @@ def download(self, engine=None, debug=False):
                     write.write(','.join(str(value) for value in newvalues))
                 write.close()
                 read.close()
-
             engine.auto_create_table(Table("weather", pk="RouteDataId",
-                                           cleanup=Cleanup(correct_invalid_value, nulls=['NULL'])),
-                                     filename="weather_new.csv")
+                                           cleanup=self.cleanup_func_table),
+                                     filename="weather_new.csv")
```
Comment (henrykironde) [inline scripts/breed_bird_survey_50stop.py]: remove space

File: scripts/la_selva_trees.py [hunk 4]
```diff
@@ -1,22 +1,30 @@
...
+from pkg_resources import parse_version
 
 
 class main(Script):
     def __init__(self):
         Script.__init__(self,
-                        tables={'trees': Table('trees', cleanup=Cleanup(correct_invalid_value, nulls=[-999]))},
-                        name="Tree growth, mortality, physical condition - Clark, 2006",
-                        tags=['plants', 'time-series'],
+                        tables={'trees': Table('trees', cleanup=Cleanup(correct_invalid_value, missingValues=[-999]))},
```
Comment (henrykironde) [inline scripts/la_selva_trees.py]: it would be nice if you clean up some of these files in this was 

```python
class main(Script):
    def __init__(self, **kwargs):
        Script.__init__(self, **kwargs)
        self.title="Gulf of Maine intertidal density/cover (Petraitis et al. 2008)",
        self.citation="Peter S. Petraitis, Harrison Liu, and Erika C. Rhile. 2008. Densities and cover data for intertidal organisms in the Gulf of Maine, USA, from 2003 to 2007. Ecology 89:588.",
        self.name="intertidal-abund-me",
        self.ref="https://figshare.com/collections/DENSITIES_AND_COVER_DATA_FOR_INTERTIDAL_ORGANISMS_IN_THE_GULF_OF_MAINE_USA_FROM_2003_TO_2007/3300200",
        self.description="The data set provides access to data on densities and percent cover in the 60 experimental plots from 2003 to 2007 and to update data from 1996 to 2002 that are already published in Ecological Archives.It includes densities of mussels, an herbivorous limpet, herbivorous snails, a predatory snail, a barnacle , and fucoid algae and percent cover by mussels, barnacles, fucoids, and other sessile organisms.",
        self.retriever_minimum_version='2.0.dev',
        self.version='1.4.0',
        self.urls={"main": "https://ndownloader.figshare.com/files/5600831"},
        self.tables={"main": Table("main", cleanup=Cleanup(correct_invalid_value, missingValues=[-999.9]))}

        if parse_version(VERSION) <= parse_version("2.0.0"):
            self.shortname = self.name
            self.name = self.title
            self.tags = self.keywords
            self.tables = {"main": Table("main", cleanup=Cleanup(correct_invalid_value, nulls=[-999.9]))}
```

especially the ones where you to clean up `cleanup=Cleanup(correct_invalid_value, nulls=[-999.9]))}`

File: scripts/plant_life_hist_eu.py [hunk 5]
```diff
@@ -43,9 +43,17 @@ def __init__(self, **kwargs):
             "Terminal_velocity": "http://www.uni-oldenburg.de/fileadmin/user_upload/biologie/ag/landeco/download/LEDA/Data_files/TV.txt",
         }
         self.citation = "KLEYER, M., BEKKER, R.M., KNEVEL, I.C., BAKKER, J.P, THOMPSON, K., SONNENSCHEIN, M., POSCHLOD, P., VAN GROENENDAEL, J.M., KLIMES, L., KLIMESOVA, J., KLOTZ, S., RUSCH, G.M., HERMY, M., ADRIAENS, D., BOEDELTJE, G., BOSSUYT, B., DANNEMANN, A., ENDELS, P., GoeTZENBERGER, L., HODGSON, J.G., JACKEL, A-K., KueHN, I., KUNZMANN, D., OZINGA, W.A., RoeMERMANN, C., STADLER, M., SCHLEGELMILCH, J., STEENDAM, H.J., TACKENBERG, O., WILMANN, B., CORNELISSEN, J.H.C., ERIKSSON, O., GARNIER, E., PECO, B. (2008): The LEDA Traitbase: A database of life-history traits of Northwest European flora. Journal of Ecology 96: 1266-1274"
-        self.tags = ['plants', 'observational']
+        self.keywords = ['plants', 'observational']
         self.description = "The LEDA Traitbase provides information on plant traits that describe three key features of plant dynamics: persistence, regeneration and dispersal. "
 
+        if parse_version(VERSION) <= parse_version("2.0.0"):
```
Comment (henrykironde) [inline scripts/plant_life_hist_eu.py]: missing imports

**Comment by jainamritanshu** [comment]
@henrykironde thank you for helping me with this. I have updated the changes. Kindly have a look

**Comment by henrykironde** [comment]
run` python  version.py`

**Comment by jainamritanshu** [comment]
@henrykironde done.

**Comment by henrykironde** [comment]
It looks like some `.py` scripts are not changed. I expect all .py files to have changes

**Comment by jainamritanshu** [comment]
@henrykironde I am sorry, I guess a missed a few. Have done the required changes.

File: scripts/gentry_forest_transects.py [hunk 6]
```diff
@@ -14,23 +14,22 @@
...
+        self.name = "gentry-forest-transects"
         self.retriever_minimum_version = '2.0.dev'
-        self.version = '1.3.1'
+        self.version = '1.4.0'
         self.urls = {"stems": "http://www.mobot.org/mobot/gentry/123/all_Excel.zip",
                      "sites": "https://ndownloader.figshare.com/files/5515373",
                      "species": "",
                      "counts": ""}
-        self.tags = ["plants", "global-scale", "observational"]
+        self.keywords = ["plants", "global-scale", "observational"]
```
Comment (henrykironde) [inline scripts/gentry_forest_transects.py]: @jainamritanshu  we are missing something in this script.

File: scripts/forest_inventory_analysis.py [hunk 7]
```diff
@@ -16,14 +16,14 @@
...
+        self.title = "Forest Inventory and Analysis"
+        self.name = "forest-inventory-analysis"
         self.retriever_minimum_version = '2.0.dev'
-        self.version = '1.3.1'
+        self.version = '1.4.0'
         self.ref = "http://fia.fs.fed.us/"
         self.urls = {"main": "https://apps.fs.usda.gov/fiadb-downloads/CSV/",
                      'species': 'https://apps.fs.usda.gov/fiadb-downloads/CSV/REF_SPECIES.csv'}
-        self.tags = ["plants", "continental-scale", "observational"]
+        self.keywords = ["plants", "continental-scale", "observational"]
```
Comment (henrykironde) [inline scripts/forest_inventory_analysis.py]: missing some changes for this script  if` parse_version(VERSION) <= parse_version("2.0.0")`

Comment (jainamritanshu) [inline scripts/forest_inventory_analysis.py]: @henrykironde I have added the parse_version feature in the script.

Comment (henrykironde) [inline scripts/forest_inventory_analysis.py]: there are more missing changes.
`scripts/MammalSuperTree.py`

Comment (jainamritanshu) [inline scripts/forest_inventory_analysis.py]: @henrykironde I have made some changes in the script `MammalSuperTree.py`, it doesn't look much clean. I had put in some thought, unfortunately couldn't come with something better. Kindly have a look and suggest if something better comes up.

Comment (henrykironde) [inline scripts/forest_inventory_analysis.py]: I have emailed you structure, revise it

**Comment by jainamritanshu** [comment]
@henrykironde thanks for the suggestion. I have made the respective changes.

**Comment by ethanwhite** [comment]
This appears to be failing locally for me when testing against 2.0.0:

```
$ pip install retriever --upgrade #This gets the 2.0.0 version from pypi
$ retriever reset all #Remove any cached scripts
$ retriever ls #Run this from the root directory of the repo with this branch checked out so it uses the scripts from this PR
Failed to load script: gentry_forest_transects (scripts)
Exception: name 'parse_version' is not defined 
Failed to load script: MammalSuperTree (scripts)
Exception: name 'Script' is not defined 
Failed to load script: forest_inventory_analysis (scripts)
Exception: name 'parse_version' is not defined 
Failed to load script: gentry_forest_transects (scripts)
Exception: name 'parse_version' is not defined 
Failed to load script: MammalSuperTree (scripts)
Exception: name 'Script' is not defined 
Failed to load script: forest_inventory_analysis (scripts)
Exception: name 'parse_version' is not defined 
Available datasets : 58

abalone-age               breast-cancer-wi              fish-parasite-hosts    intertidal-abund-me  mammal-metabolic-rate  NPN                 plant-occur-oosting        prism-climate       vertnet-fishes    
amniote-life-hist         breed-bird-survey             forest-biomass-china   iris                 mapped-plant-quads-co  nyc-tree-count      plant-taxonomy-us          tree-demog-wghats   vertnet-mammals   
antarctic-breed-bird      breed-bird-survey-50stop      forest-fires-portugal  leaf-herbivory       mapped-plant-quads-id  pantheria           poker-hands                veg-plots-sdl       vertnet-reptiles  
aquatic-animal-excretion  butterfly-population-network  forest-plots-michigan  mammal-community-db  mapped-plant-quads-ks  phytoplankton-size  portal                     vertnet             wine-composition  
bioclim                   car-eval                      forest-plots-wghats    mammal-diet          mapped-plant-quads-mt  plant-comp-ok       portal-dev                 vertnet-amphibians  wine-quality      
biomass-allometry-db      community-abundance-misc      fray-jorge-ecology     mammal-life-hist     mt-st-helens-veg       plant-life-hist-eu  predator-prey-size-marine  vertnet-birds       wood-density      
bird-size                 elton-traits                  home-ranges            mammal-masses
```

If you can test to see if everything is working using this approach, modify it until everything appears to work normally both for the source code in this branch and for `2.0.0`, and then either update this PR or let us know if you're running into issues running down a fix for something, then we can go from there.

Here's the instructions for testing against `2.0.0` pulled out from the output above:

```
# Get the 2.0.0 version from pypi
pip install retriever --upgrade

# Remove any cached scripts
retriever reset all

# Try retriever commands from the root directory of the repo with this branch checked out 
retriever ls
retriever install csv gentry-forest-transects
```

**Comment by zhangcandrew** [comment]
Hey @jainamritanshu, it looks like you're just missing some imports in these scripts: 

gentry_forest_transects - parse version
- `from pkg_resources import parse_version`
- `from retriever import VERSION`

MammalSuperTree - Script
-  `from retriever.lib.templates import Script`

forest_inventory_analysis - parse version
- `from pkg_resources import parse_version`
- `from retriever import VERSION`

I tested it locally against 2.0.0, and once those imports are put in, it seems to work fine.

**Comment by jainamritanshu** [comment]
@zhangcandrew @ethanwhite I got understood the errors, thanks. I had updated the PR just forgot to refer it in the comment. It works fine now. Kindly have a look, and tell me if I am still missing something.

**Comment by henrykironde** [comment]
@jainamritanshu Could you run some simple test on this PR just as Ethan mentioned the process of testing on version 2.0.0 above
Go ahead and follow the instructions and provide me with results of `retriever -v` and `retriever ls`

After that run 
```
pip uninstall retriever(uninstall ) repeat this until you are sure that there is no installed version
pip install . -u 
```
after the above provide me with the results of 
```
retriever -v 
retriever ls
```

**Comment by jainamritanshu** [comment]
for `retriever ls`
```
Available datasets : 61

abalone-age                   mammal-super-tree          
amniote-life-hist             mapped-plant-quads-co      
antarctic-breed-bird          mapped-plant-quads-id      
aquatic-animal-excretion      mapped-plant-quads-ks      
bioclim                       mapped-plant-quads-mt      
biomass-allometry-db          mt-st-helens-veg           
bird-size                     NPN                        
breast-cancer-wi              nyc-tree-count             
breed-bird-survey             pantheria                  
breed-bird-survey-50stop      phytoplankton-size         
butterfly-population-network  plant-comp-ok              
car-eval                      plant-life-hist-eu         
community-abundance-misc      plant-occur-oosting        
elton-traits                  plant-taxonomy-us          
fish-parasite-hosts           poker-hands                
forest-biomass-china          portal                     
forest-fires-portugal         portal-dev                 
forest-inventory-analysis     predator-prey-size-marine  
forest-plots-michigan         prism-climate              
forest-plots-wghats           tree-demog-wghats          
fray-jorge-ecology            veg-plots-sdl              
gentry-forest-transects       vertnet                    
home-ranges                   vertnet-amphibians         
intertidal-abund-me           vertnet-birds              
iris                          vertnet-fishes             
leaf-herbivory                vertnet-mammals            
mammal-community-db           vertnet-reptiles           
mammal-diet                   wine-composition           
mammal-life-hist              wine-quality               
mammal-masses                 wood-density               
mammal-metabolic-rate
```

for `retriever -v`
```
v2.1.dev
```

**Comment by henrykironde** [comment]
@jainamritanshu Thanks for the test on version v2.1.dev. The scripts seem to be all loading and working well.

You still need to test version 2.0.0, you need to uninstall any version and test on the old version of 2.0.0
To do that you will follow these steps.

Checkout your branch or make sure you are on the branch `update-internals-final`.

- [ ] Uninstall any retriever version by running `pip uninstall retriever`, repeat this until you are sure there is no retriever installed

- [ ] Install retriever version 2.0.0, Run `pip install retriever`

- [ ] Test the version of retriever `retriever install -v`. This should return `v2.0.0`

- [ ] Run `retriever reset all` and `git clean -fx` this will remove any cached scripts

- [ ] Run `retriever ls` this should have the same results as the ones you have provide above for version v2.1.dev 

- [ ] Share your findings or let me know when you get stuck

To sum up, since you have tested your branch `update-internals-final`, which should be v2.1.dev, The scripts in this branch should still work for users who already have retriever and are still running the old version.

**Comment by jainamritanshu** [comment]
- [x] Uninstall any retriever version by running `pip uninstall retriever`, repeat this until you are sure there is no retriever installed

- [x] Install retriever `version 2.0.0`, Run `pip install retriever`

- [x] Test the version of retriever `retriever install -v`. This should return v2.0.0

- [x] Run `retriever reset all` and `git clean -fx` this will remove any cached scripts

- [x] Run retriever ls this should have the same results as the ones you have provide above for version v2.1.dev

- [x] Share your findings or let me know when you get stuck

retriever is working fine with version 2.0.0 as well. `retriever ls` returned
```
Available datasets : 61

abalone-age                   mammal-super-tree          
amniote-life-hist             mapped-plant-quads-co      
antarctic-breed-bird          mapped-plant-quads-id      
aquatic-animal-excretion      mapped-plant-quads-ks      
bioclim                       mapped-plant-quads-mt      
biomass-allometry-db          mt-st-helens-veg           
bird-size                     NPN                        
breast-cancer-wi              nyc-tree-count             
breed-bird-survey             pantheria                  
breed-bird-survey-50stop      phytoplankton-size         
butterfly-population-network  plant-comp-ok              
car-eval                      plant-life-hist-eu         
community-abundance-misc      plant-occur-oosting        
elton-traits                  plant-taxonomy-us          
fish-parasite-hosts           poker-hands                
forest-biomass-china          portal                     
forest-fires-portugal         portal-dev                 
forest-inventory-analysis     predator-prey-size-marine  
forest-plots-michigan         prism-climate              
forest-plots-wghats           tree-demog-wghats          
fray-jorge-ecology            veg-plots-sdl              
gentry-forest-transects       vertnet                    
home-ranges                   vertnet-amphibians         
intertidal-abund-me           vertnet-birds              
iris                          vertnet-fishes             
leaf-herbivory                vertnet-mammals            
mammal-community-db           vertnet-reptiles           
mammal-diet                   wine-composition           
mammal-life-hist              wine-quality               
mammal-masses                 wood-density               
mammal-metabolic-rate 
```

File: scripts/la_selva_trees.py (hunk 4 above)
Comment (ethanwhite) [inline scripts/la_selva_trees.py]: This is basically the same comment as the one I made on `intertidal-abund-md`.

File: scripts/la_selva_trees.py [hunk 8]
```diff
@@ -1,22 +1,30 @@
...
+                        title="Tree growth, mortality, physical condition - Clark, 2006",
+                        keywords=['plants', 'time-series'],
                         urls={'trees': 'https://ndownloader.figshare.com/files/5597693'},
-                        shortname="la-selva-trees",
+                        sname="la-selva-trees",
```
Comment (ethanwhite) [inline scripts/la_selva_trees.py]: `name` not `sname`

File: scripts/npn.py [hunk 9]
```diff
@@ -10,26 +10,35 @@
 standard_library.install_aliases()
 from builtins import str
 
+import os
+import urllib.request, urllib.parse, urllib.error
+import zipfile
+from decimal import Decimal
```
Comment (ethanwhite) [inline scripts/npn.py]: Why are all of these new imports in this PR?

**Comment by ethanwhite** [comment]
@jainamritanshu - Looks like the tests are failing on that last commit. Can you take a look at see if you can figure out there error. If it's not clear let me know and I'll take a look

File: scripts/intertidal_abund_me.py [hunk 10]
```diff
@@ -1,21 +1,26 @@
...
-                        ref="https://figshare.com/collections/DENSITIES_AND_COVER_DATA_FOR_INTERTIDAL_ORGANISMS_IN_THE_GULF_OF_MAINE_USA_FROM_2003_TO_2007/3300200",
-                        description="The data set provides access to data on densities and percent cover in the 60 experimental plots from 2003 to 2007 and to update data from 1996 to 2002 that are already published in Ecological Archives.It includes densities of mussels, an herbivorous limpet, herbivorous snails, a predatory snail, a barnacle , and fucoid algae and percent cover by mussels, barnacles, fucoids, and other sessile organisms.",
-                        retriever_minimum_version='2.0.dev',
-                        version='1.4.0',
-                        urls={"main": "https://ndownloader.figshare.com/files/5600831"},
-                        tables={"main": Table("main", cleanup=Cleanup(correct_invalid_value, nulls=[-999.9]))}
-                        )
+    def __init__(self, **kwargs):
+        Script.__init__(self, **kwargs)
+        self.title="Gulf of Maine intertidal density/cover (Petraitis et al. 2008)",
```
Comment (henrykironde) [inline scripts/intertidal_abund_me.py]: Remove the commas at the end ` ,`

File: scripts/la_selva_trees.py [hunk 11]
```diff
@@ -1,21 +1,29 @@
...
-                        urls={'trees': 'https://ndownloader.figshare.com/files/5597693'},
-                        shortname="la-selva-trees",
-                        description="The data set helps to examine the post-establishment ecology of 10 species of tropical wet forest trees selected to span a range of predicted life history patterns at the La Selva Biological Station in Costa Rica.",
-                        ref="https://doi.org/10.6084/m9.figshare.c.3299324.v1",
-                        retriever_minimum_version= "2.0.dev",
-                        version='1.3.0',
-                        citation="David B. Clark and Deborah A. Clark. 2006. Tree growth, mortality, physical condition, and microsite in an old-growth lowland tropical rain forest. Ecology 87:2132.")
+    def __init__(self, **kwargs):
+        Script.__init__(self, **kwargs)
+        self.tables={'trees': Table('trees', cleanup=Cleanup(correct_invalid_value, missingValues=[-999]))},
```
Comment (henrykironde) [inline scripts/la_selva_trees.py]: Remove the commas at the end ` ,`

**Comment by henrykironde** [comment]
Update the version number in `_version.py`.

**Comment by henrykironde** [comment]
And amend the commits 👍

**Comment by jainamritanshu** [comment]
@henrykironde what should be the version number in `_version.py`? Also travis-ci is failing for version 2.7

**Comment by henrykironde** [comment]
2.1.dev

**Comment by jainamritanshu** [comment]
the version is already 2.1.dev I guess

**Comment by henrykironde** [comment]
Could you please make it one commit message thanks.

**Comment by jainamritanshu** [comment]
@henrykironde Done! Kindly check once

File: version.txt [hunk 12]
```diff
@@ -1,63 +1,63 @@
...
 mammal_masses.json,1.3.0
 mammal_metabolic_rate.json,1.2.0
-MammalSuperTree.py,1.1.0
+MammalSuperTree.py,1.2.0
 mapped_plant_quads_co.json,1.2.0
 mapped_plant_quads_id.json,1.2.0
 mapped_plant_quads_ks.json,1.2.0
 mapped_plant_quads_mt.json,1.2.0
 mt_st_helens_veg.json,1.2.0
 npn.py,2.0.0
```
Comment (henrykironde) [inline version.txt]: go back and update the version number for this scripts

File: docs/scripts.rst [hunk 13]
```diff
@@ -58,7 +58,7 @@ table.
   Add Table? (y/N): y
   table-name: species
   table-url: http://esapubs.org/archive/ecol/E084/093/Mammal_lifehistories_v2.txt
-  missingValues (separated by ';'):
+  missing-values (separated by ';'):
```
Comment (henrykironde) [inline docs/scripts.rst]: According to a test run `retriever  new_json`, this looks like it should be `missing values` instead of `missing-values`, so I guess we can go a head and update that correctly.

**Comment by jainamritanshu** [comment]
Sure @henrykironde :smile: There are already 2 commits as @ethanwhite told me to, 1 commit for changes in scripts, and one for the general environment of retriever mainly in `retriever/lib`. So should I amend the changes?

File: version.txt (hunk 12 above)
Comment (henrykironde) [inline version.txt]: The best way to update version.txt file is clean up your repository. Run `git clean -fx` and then `retriever reset all`. This gives it a clean environment to run`python version.py`

**Comment by henrykironde** [comment]
wait

**Comment by henrykironde** [comment]
In that case create the 3rd as a clean up. This is because the PR is big and at this moment I have tested the two commits, we should be able to track any changes to avoid spending more time on reviewing everything critically including the past working patches.

**Comment by jainamritanshu** [comment]
Sure @henrykironde Will do that right away

**Comment by henrykironde** [comment]
@ethanwhite, Could you take a look at the state of this PR.
I think I have exhausted all the needed changes I could find.
LGTM and if there are any required changes, will be happy to look through them. 
Python2 fails are solved in a separate PR.

File: scripts/breed_bird_survey.py [hunk 14]
```diff
@@ -36,6 +37,16 @@ def __init__(self, **kwargs):
...
 
+        if parse_version(VERSION) <= parse_version("2.0.0"):
+            self.shortname = self.name
+            self.name = self.title
+            self.tags = self.keywords
+            self.cleanup_func_table = Cleanup(correct_invalid_value, nulls=['NULL'])
+            self.cleanup_func_clean = Cleanup(correct_invalid_value, nulls = ['*'])
+        else:
+            self.cleanup_func_table = Cleanup(correct_invalid_value, missingValues=['NULL'])
+            self.cleanup_func_clean = Cleanup(correct_invalid_value, missingValues = ['*'])
```
Comment (ethanwhite) [inline scripts/breed_bird_survey.py]: Should these be `missing_values`?

File: scripts/breed_bird_survey_50stop.py [hunk 15]
```diff
@@ -37,6 +38,16 @@ def __init__(self, **kwargs):
...
 
+        if parse_version(VERSION) <= parse_version("2.0.0"):
+            self.shortname = self.name
+            self.name = self.title
+            self.tags = self.keywords
+            self.cleanup_func_table = Cleanup(correct_invalid_value, nulls=['NULL'])
+            self.cleanup_func_clean = Cleanup(correct_invalid_value, nulls = ['*'])
+        else:
+            self.cleanup_func_table = Cleanup(correct_invalid_value, missingValues=['NULL'])
+            self.cleanup_func_clean = Cleanup(correct_invalid_value, missingValues = ['*'])
```
Comment (ethanwhite) [inline scripts/breed_bird_survey_50stop.py]: Should these be `missing_values`?

File: scripts/intertidal_abund_me.py [hunk 16]
```diff
@@ -1,21 +1,26 @@
...
+        self.title="Gulf of Maine intertidal density/cover (Petraitis et al. 2008)"
+        self.citation="Peter S. Petraitis, Harrison Liu, and Erika C. Rhile. 2008. Densities and cover data for intertidal organisms in the Gulf of Maine, USA, from 2003 to 2007. Ecology 89:588."
+        self.name="intertidal-abund-me"
+        self.ref="https://figshare.com/collections/DENSITIES_AND_COVER_DATA_FOR_INTERTIDAL_ORGANISMS_IN_THE_GULF_OF_MAINE_USA_FROM_2003_TO_2007/3300200"
+        self.description="The data set provides access to data on densities and percent cover in the 60 experimental plots from 2003 to 2007 and to update data from 1996 to 2002 that are already published in Ecological Archives.It includes densities of mussels, an herbivorous limpet, herbivorous snails, a predatory snail, a barnacle , and fucoid algae and percent cover by mussels, barnacles, fucoids, and other sessile organisms."
+        self.retriever_minimum_version='2.0.dev'
+        self.version='1.5.0'
+        self.urls={"main": "https://ndownloader.figshare.com/files/5600831"}
+        self.tables={"main": Table("main", cleanup=Cleanup(correct_invalid_value, missingValues=[-999.9]))}
```
Comment (ethanwhite) [inline scripts/intertidal_abund_me.py]: Should this be `missing_values`?

We should put this line inside an else statement on the `parse_version` conditional like it is in the other scripts.

File: scripts/la_selva_trees.py [hunk 17]
```diff
@@ -1,21 +1,29 @@
...
-                        urls={'trees': 'https://ndownloader.figshare.com/files/5597693'},
-                        shortname="la-selva-trees",
-                        description="The data set helps to examine the post-establishment ecology of 10 species of tropical wet forest trees selected to span a range of predicted life history patterns at the La Selva Biological Station in Costa Rica.",
-                        ref="https://doi.org/10.6084/m9.figshare.c.3299324.v1",
-                        retriever_minimum_version= "2.0.dev",
-                        version='1.3.0',
-                        citation="David B. Clark and Deborah A. Clark. 2006. Tree growth, mortality, physical condition, and microsite in an old-growth lowland tropical rain forest. Ecology 87:2132.")
+    def __init__(self, **kwargs):
+        Script.__init__(self, **kwargs)
+        self.tables={'trees': Table('trees', cleanup=Cleanup(correct_invalid_value, missingValues=[-999]))}
```
Comment (ethanwhite) [inline scripts/la_selva_trees.py]: Should this be `missing_values`.

We should use the same structure in this script as in the others, where place this inside an `else` as part of the `parse_version` conditional.

File: scripts/plant_life_hist_eu.py [hunk 18]
```diff
@@ -67,8 +76,7 @@ def download(self, engine=None, debug=False):
             file_block.close()
             new_data.close()
             self.engine.auto_create_table(Table(key,
-                                                cleanup=Cleanup(correct_invalid_value,
-                                                                nulls=[-999.9])),filename=str("new" + key))
+                                                cleanup=cleanup_func_table),filename=str("new" + key))
```
Comment (ethanwhite) [inline scripts/plant_life_hist_eu.py]: Missing space after the `,`

**Comment by jainamritanshu** [comment]
@ethanwhite thanks for reviewing. I missed a few replacements of `missingValues` in python scripts as I realize now and it should be `missing_values` as you said and according the variables in `lib/cleanup.py`. The cleanup function sure looks much cleaner by following your suggestion in the scripts. I have made a commit with all these changes, kindly give it a read.

File: version.txt [hunk 19]
```diff
@@ -1,4 +1,4 @@
-v2.1.dev
+v2.0.0
```
Comment (henrykironde) [inline version.txt]: This should not be changed

File: scripts/intertidal_abund_me.py [hunk 20]
```diff
@@ -15,12 +15,13 @@ def __init__(self, **kwargs):
...
         self.urls={"main": "https://ndownloader.figshare.com/files/5600831"}
-        self.tables={"main": Table("main", cleanup=Cleanup(correct_invalid_value, missingValues=[-999.9]))}
+        self.cleanup_func_table = Cleanup(correct_invalid_value, missing_values=[-999.9])
+        self.tables={"main": Table("main", cleanup=self.cleanup_func_table)}
                         
         if parse_version(VERSION) <= parse_version("2.0.0"):
             self.shortname = self.name
             self.name = self.title
-            self.tables = {"main": Table("main", cleanup=Cleanup(correct_invalid_value, nulls=[-999.9]))}
+            self.cleanup_func_table = Cleanup(correct_invalid_value, nulls=[-999.9])
```
Comment (ethanwhite) [inline scripts/intertidal_abund_me.py]: I think this needs to be something like:

```
        self.urls={"main": "https://ndownloader.figshare.com/files/5600831"}

        if parse_version(VERSION) <= parse_version("2.0.0"):
            self.shortname = self.name
            self.name = self.title
            self.cleanup_func_table = Cleanup(correct_invalid_value, nulls=[-999.9])
        else:
            self.cleanup_func_table = Cleanup(correct_invalid_value, missing_values=[-999.9])
        self.tables={"main": Table("main", cleanup=self.cleanup_func_table)}
```

As it is I don't think `self.tables` would be updated when you change `self.cleanup_func_table` in the conditional. The same change applies to `la-selva-trees`.

**Comment by jainamritanshu** [comment]
@ethanwhite I have made the required changes with a slight variation. Instead of
```
        if parse_version(VERSION) <= parse_version("2.0.0"):
            self.shortname = self.name
            self.name = self.title
            self.cleanup_func_table = Cleanup(correct_invalid_value, nulls=[-999.9])
        else:
            self.cleanup_func_table = Cleanup(correct_invalid_value, missing_values=[-999.9])
        self.tables={"main": Table("main", cleanup=self.cleanup_func_table)}
```
I did - 
```
        self.cleanup_func_table = Cleanup(correct_invalid_value, missing_values=[-999.9])
        if parse_version(VERSION) <= parse_version("2.0.0"):
            self.shortname = self.name
            self.name = self.title
            self.cleanup_func_table = Cleanup(correct_invalid_value, nulls=[-999.9])
        self.tables={"main": Table("main", cleanup=self.cleanup_func_table)}
```

**Comment by ethanwhite** [comment]
@jainamritanshu That works. It looks like you accidentally reverted the version number again.

**Comment by jainamritanshu** [comment]
@ethanwhite fixed it

File: scripts/la_selva_trees.py [hunk 21]
```diff
@@ -8,7 +8,7 @@
 class main(Script):
     def __init__(self, **kwargs):
         Script.__init__(self, **kwargs)
-        self.tables={'trees': Table('trees', cleanup=Cleanup(correct_invalid_value, missingValues=[-999]))}
+        self.cleanup_func_table = Cleanup(correct_invalid_value, missing_values=[-999.9])
```
Comment (ethanwhite) [inline scripts/la_selva_trees.py]: Was there a reason to change this from `-999` to `-999.9`? Looks like this broke the current null value handling (at least on 2.0.0).

File: scripts/la_selva_trees.py [hunk 22]
```diff
@@ -23,7 +23,8 @@ def __init__(self, **kwargs):
             self.shortname = self.name
             self.name = self.title
             self.tags = self.keywords
-            self.tables = {"trees": Table("trees", cleanup=Cleanup(correct_invalid_value, nulls=[-999.9]))}
+            self.cleanup_func_table = Cleanup(correct_invalid_value, nulls=[-999.9])
```
Comment (ethanwhite) [inline scripts/la_selva_trees.py]: Assuming the above line should be `-999` this one should be as well.

File: scripts/la_selva_trees.py (hunk 21 above)
Comment (jainamritanshu) [inline scripts/la_selva_trees.py]: I guess it got changed accidentally. I will fix it right away

**Comment by henrykironde** [comment]
Works fine for the group of tests I have conducted on both Versions, 2.0.0 and 2.1.dev.

Tested with a few scripts that had the null values and works as expected. 
pantheriatest dataset needs an update of the null values. We do not need to include it in this PR, I have created a separate issue for it.

**Comment by henrykironde** [comment]
@ethanwhite LGTM

**Comment by ethanwhite** [comment]
Excellent work @jainamritanshu! Thanks so much for sticking with this one. I know it took a while and a lot of changes, but it's a nice improvement that will really help with maintainability going forward. In it goes.
